### PR TITLE
FOGL-1033 storage client exceptions test

### PR DESCRIPTION
**Pull request by praveen-garg**
> $ pytest tests/unit/python/foglamp/common/storage_client/test_exceptions.py 

```
======================================================= test session starts ========================================================
platform linux -- Python 3.5.2, pytest-3.4.0, py-1.5.2, pluggy-0.6.0
rootdir: /home/foglamp/foglamp/FogLAMP, inifile:
plugins: mock-1.6.0, cov-2.5.1, asyncio-0.8.0, allure-adaptor-1.7.10, aiohttp-0.3.0
collected 9 items                                                                                                                  

tests/unit/python/foglamp/common/storage_client/test_exceptions.py .........                                                 [100%]

===================================================== 9 passed in 0.02 seconds =====================================================

```

> [FOGL-1033] $ pytest tests/unit/python/

```
============================================== 147 passed, 1 skipped in 1.22 seconds ===============================================
```

### CODE REVIEW

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("common", "storage_client")
+class TestStorageClientExceptions:
+
+    def test_init_StorageClientException(self):
+
+        with pytest.raises(Exception) as excinfo:
+            raise StorageClientException()
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: If no arg passed, remove parentheses, works well without it

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            assert ex.__class__ is StorageClientException
+            assert issubclass(ex.__class__, Exception)
+            assert 40 == ex.code
+            assert ex.message is None
+
+    def test_init_args_StorageClientException(self):
+        with pytest.raises(Exception) as excinfo:
+            raise StorageClientException(code=11, message="foo")
+        assert excinfo.type is StorageClientException
+        assert issubclass(excinfo.type, Exception)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: Same assertions called below? If that is just for demo purpose for pytest raise wrapper, can we comment it out and have single place of all assertions?

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            assert ex.__class__ is StorageClientException
+            assert issubclass(ex.__class__, Exception)
+            assert 11 == ex.code
+            assert "foo" == ex.message
+
+    def test_BadRequest(self):
+        with pytest.raises(Exception) as excinfo:
+            raise BadRequest()
+        assert excinfo.type is BadRequest
+        assert issubclass(excinfo.type, StorageClientException)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: Same as above

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            assert ex.__class__ is BadRequest
+            assert issubclass(ex.__class__, StorageClientException)
+            assert 400 == ex.code
+            assert "Bad request" == ex.message
+
+    def test_StorageServiceUnavailable(self):
+        with pytest.raises(Exception) as excinfo:
+            raise StorageServiceUnavailable()
+        assert excinfo.type is StorageServiceUnavailable
+        assert issubclass(excinfo.type, StorageClientException)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: Same as above

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            assert ex.__class__ is StorageServiceUnavailable
+            assert issubclass(ex.__class__, StorageClientException)
+            assert 503 == ex.code
+            assert "Storage service is unavailable" == ex.message
+
+    def test_InvalidServiceInstance(self):
+        with pytest.raises(Exception) as excinfo:
+            raise InvalidServiceInstance()
+        assert excinfo.type is InvalidServiceInstance
+        assert issubclass(excinfo.type, StorageClientException)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: Same as above

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+            assert ex.__class__ is InvalidServiceInstance
+            assert issubclass(ex.__class__, StorageClientException)
+            assert 502 == ex.code
+            assert "Storage client needs a valid *FogLAMP storage* micro-service instance" == ex.message
+
+    def test_InvalidReadingsPurgeFlagParameters(self):
+        with pytest.raises(Exception) as excinfo:
+            raise InvalidReadingsPurgeFlagParameters()
+        assert excinfo.type is InvalidReadingsPurgeFlagParameters
+        assert issubclass(excinfo.type, BadRequest)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: Same as above

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+            assert ex.__class__ is InvalidReadingsPurgeFlagParameters
+            assert issubclass(ex.__class__, BadRequest)
+            assert 400 == ex.code
+            assert "Purge flag valid options are retain or purge only" == ex.message
+
+    def test_PurgeOneOfAgeAndSize(self):
+        with pytest.raises(Exception) as excinfo:
+            raise PurgeOneOfAgeAndSize()
+        assert excinfo.type is PurgeOneOfAgeAndSize
+        assert issubclass(excinfo.type, BadRequest)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: Same as above

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+            assert ex.__class__ is PurgeOneOfAgeAndSize
+            assert issubclass(ex.__class__, BadRequest)
+            assert 400 == ex.code
+            assert "Purge must specify one of age or size" == ex.message
+
+    def test_PurgeOnlyOneOfAgeAndSize(self):
+        with pytest.raises(Exception) as excinfo:
+            raise PurgeOnlyOneOfAgeAndSize()
+        assert excinfo.type is PurgeOnlyOneOfAgeAndSize
+        assert issubclass(excinfo.type, BadRequest)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: Same as above

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py (hunk 1 above)
Comment (praveen-garg) [reply tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: this is an explicit class for internal use to subclass. Hence an explicit class test

File: tests/unit/python/foglamp/common/storage_client/test_exceptions.py (hunk 2 above)
Comment (praveen-garg) [reply tests/unit/python/foglamp/common/storage_client/test_exceptions.py]: > # pytest raises wrapper allow only type, value and traceback info

So yes, we can say its for that explanation. I don;t think we should remove these or comment

**Pull request merged**